OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -236,6 +243,26 @@ void ITick.Tick(Actor self)
 		}
 
 		protected virtual void Tick(Actor self)
+		{
+			// PERF: Avoid LINQ when checking whether all production traits are disabled/paused
+			var anyEnabledProduction = false;
+			var anyUnpausedProduction = false;
+			foreach (var p in productionTraits)
+			{
+				if (!anyEnabledProduction)
+					anyEnabledProduction = !p.IsTraitDisabled;
+
+				if (!anyUnpausedProduction)
+					anyUnpausedProduction = !p.IsTraitPaused;
+			}
+
+			if (!anyEnabledProduction)
+				ClearQueue();
+
+			QueueTick(self, !anyUnpausedProduction);
+		}
+
+		protected virtual void QueueTick(Actor self, bool allProductionPaused)
```

Comment:
Can we go with `TickQueue` please.

Reply:
`TickInner` would be more in line with other parts of the codebase.